src/term/util.rs

```diff
@@ -0,0 +1,56 @@
+// TODO: Find a way to implement Clone for ChildrenIter
+
+/// Iterator for tree children
+pub enum ChildrenIter<'a, T: 'a> {
+  Zero(std::array::IntoIter<T, 0>),
+  One(std::array::IntoIter<T, 1>),
+  Two(std::array::IntoIter<T, 2>),
+  Many(Box<dyn DoubleEndedIterator<Item = T> + 'a>),
+}
```

Comment:
This reminds me of: https://xkcd.com/764/

Primitive cultures develop ~~sesame street~~ iterators 😆